src/Instance.ts

```diff
@@ -158,7 +158,7 @@ export class Marked {
           renderer[rendererProp] = (...args: unknown[]) => {
             if (!pack.useNewRenderer) {
               // TODO: Remove this in next major version
-              rendererFunc = this.#convertRendererFunction(rendererFunc, rendererProp, renderer) as GenericRendererFunction;
+              rendererFunc = this.#convertRendererFunction(pack.renderer![rendererProp] as GenericRendererFunction, rendererProp, renderer) as GenericRendererFunction;
             }
```

Comment:
I feel like moving this out of the `renderer[rendererProp]` function would be better. There is no reason we have to assign rendererFunc every time the function is called.

Reply:
yes, moving this out of make sense.